### PR TITLE
Add test macro wrapper and fix spin_bvh test

### PR DESCRIPTION
**Pull request by white238**
#349 

* Adds a common `axom_add_test` macro, this allows us to do project-wide changes to all our tests in one place
* Fix spin_bvh test by setting environment variable `CUDA_VISIBLE_DEVICES`

### CODE REVIEW

File: src/cmake/AxomMacros.cmake [hunk 1]
```diff
@@ -115,6 +115,46 @@ macro(axom_add_component)
...
 
 
+##------------------------------------------------------------------------------
+## axom_add_test(NAME            [name]
+##               COMMAND         [command] 
+##               NUM_MPI_TASKS   [n]
+##               NUM_OMP_THREADS [n]
+##               CONFIGURATIONS  [config1 [config2...]])
+##
+## Wrapper around axom_add_test() that handles functionality that Axom applies to all
```
Comment (agcapps) [inline src/cmake/AxomMacros.cmake:125]: Should be "around blt_add_test() that handles functionality"

**Review by agcapps — APPROVED** [review]
Thanks @white238

File: src/cmake/AxomMacros.cmake (hunk 1 above)
Comment (white238) [reply src/cmake/AxomMacros.cmake:125]: yepp, missed this after the sed script. Thanks!

File: src/axom/spin/tests/CMakeLists.txt [hunk 2]
```diff
@@ -33,38 +33,46 @@ blt_list_append( TO spin_depends ELEMENTS cuda IF ${ENABLE_CUDA} )
...
+      COMMAND         spin_bvh_test
       NUM_OMP_THREADS 4
-     )
+      )
+
+    # fix for rzmanta of following runtime error:
+    #   CUDAassert: peer access is not supported between these two devices /unimportant/path/raja-0.12.1/include/RAJA/policy/cuda/sort.hpp 341
+    #   unknown file: Failure
+    #   C++ exception with description "CUDAassert" thrown in the test body.
+    set_property(TEST spin_bvh
```
Comment (kennyweiss) [inline src/axom/spin/tests/CMakeLists.txt:73]: 👍

Comment (kennyweiss) [reply src/axom/spin/tests/CMakeLists.txt:73]: Would it be worthwhile to fold `ENVIRONMENT` into `blt_add_test` or `axom_add_test`?

(Clearly this in not required, but might be a nice convenience?)

File: src/cmake/AxomMacros.cmake [hunk 3]
```diff
@@ -115,6 +115,46 @@ macro(axom_add_component)
...
+                 NUM_OMP_THREADS ${arg_NUM_OMP_THREADS}
+                 CONFIGURATIONS  ${arg_CONFIGURATIONS} )
+
+    ###########################################################################
+    # Newer versions of OpenMPI require OMPI_MCA_rmaps_base_oversubscribe=1
+    # to run with more tasks than actual cores
+    # Since this is an OpenMPI specific env var, it shouldn't interfere
+    # with other mpi implementations.
+    ###########################################################################
+    set_property(TEST ${arg_NAME}
```
Comment (kennyweiss) [inline src/cmake/AxomMacros.cmake:151]: 👍

File: src/docs/sphinx/dev_guide/testing.rst [hunk 4]
```diff
@@ -320,8 +320,8 @@ following items:
 
      and::
 
-       blt_add_test(NAME <example executable name>
-                    COMMAND <example executable name>)
+       axom_add_test(NAME <example executable name>
+                     COMMAND <example executable name>)
```
Comment (kennyweiss) [inline src/docs/sphinx/dev_guide/testing.rst:324]: Should we list the other possible options here?
and/or, point to the docs for `blt_add_test` so users know what's available?

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 !

File: src/axom/spin/tests/CMakeLists.txt (hunk 2 above)
Comment (white238) [reply src/axom/spin/tests/CMakeLists.txt:73]: https://github.com/LLNL/blt/issues/443

File: src/cmake/AxomMacros.cmake (hunk 3 above)
Comment (white238) [reply src/cmake/AxomMacros.cmake:151]: No idea if it did something..

File: src/docs/sphinx/dev_guide/testing.rst (hunk 4 above)
Comment (white238) [reply src/docs/sphinx/dev_guide/testing.rst:324]: This was just a bulk rename but I'd rather not just duplicate documentation

**Pull request merged**